### DISCUSSION TITLE
Downgrading from etcd to sqlite

### DISCUSSION THREAD

**Discussion by optix2000** [k3s-io/k3s#5453] [Q&A]
I understand that downgrading from etcd not sqlite is _not_ supported.

I've hit a situation where the k3s master slows down and crashes when the disks get loaded up due to the amount of writes etcd needs. Given sqlite requires far less writes, I want to somehow get back to using sqlite as the backing store with full understanding of the redundancy loss that comes from it.

My current idea is to back-up the data in etcd, convert it to sqlite, and then starting the k3s master using the new sqlite DB. I'd love any sort of of guidance on where to look to get started on this or if there are better ideas of attempting a downgrade.

Recreating the cluster isn't feasible as most pods are stateful and would require me to manually backup, map, and restore volumes to their corresponding pods.

**Comment by schmitch** [comment]
when you don't have volumes you can use velero (and if you have volumes it might work aswell), velero will backup **all** manifests, so you can just create a new k3s cluster and restore the backup there and than delete the old cluster.

**Reply by optix2000** [reply]
I considered this, but k3s uses [local-path-provisioner](https://github.com/rancher/local-path-provisioner). `local-path-provisioner` uses `hostPath` which doesn't work with Velero. See https://github.com/rancher/local-path-provisioner/issues/85

**Comment by brandond** [comment]
There is not currently any support for downgrading from etcd to sqlite. Sqlite to etcd is the only supported transition.
